### PR TITLE
MariaDB [rework security, eliminate {{ mysql_root_password }}]

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes bug:
proposed fix for #2475
### Description of changes proposed in this pull request:
don't set password for root using 'localhost', add second admin account(mariaDBadmin)
### Smoke-tested on which OS or OS's:
pending
### Mention a team member @username e.g. to help with code review:
@holta

### CODE REVIEW

File: roles/mysql/defaults/main.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ## mysql_root_password: $6$iiab51$3ICIW0CLWxxMW2a3yrHZ38ukZItD5tcadL4rWcE9D.qIGStxhh8rRsaSxoj3b.MYxI/VRDNjpzSYK/V6zkWFI0
 # mysql_root_password: fixmysql
+mysql_admin: mariaDBadmin
```
Comment (holta) [inline roles/mysql/defaults/main.yml]: Let's move this line down -- or put it in default_vars.yml ?

Either way: to respect the "# All above are set in: github.com/iiab/iiab/blob/master/vars/default_vars.yml" comment just below.

Comment (jvonau) [reply roles/mysql/defaults/main.yml]: It's just for quick testing.

**Comment by jvonau** [comment]
>iiab-admin@u20-srv:/opt/iiab/iiab$ sudo mysql -u root
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 44
Server version: 10.3.22-MariaDB-1ubuntu1 Ubuntu 20.04
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
MariaDB [(none)]> select user,host,password,plugin from mysql.user;
+--------------+-----------+-------------------------------------------+-------------+
| user         | host      | password                                  | plugin      |
+--------------+-----------+-------------------------------------------+-------------+
| root         | localhost |                                           | unix_socket |
| root         | 127.0.0.1 | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| root         | ::1       | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | 127.0.0.1 | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | ::1       | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | localhost | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
+--------------+-----------+-------------------------------------------+-------------+
6 rows in set (0.001 sec)

**Comment by jvonau** [comment]
stopped the install after stage 3 completed with the above result, ./runrole --reinstall mysql ran without issue with the same entries above. Post install below
>iiab-admin@u20-srv:/opt/iiab/iiab$ sudo mysql -u root
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 55
Server version: 10.3.22-MariaDB-1ubuntu1 Ubuntu 20.04
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
MariaDB [(none)]> select user,host,password,plugin from mysql.user;
+--------------+-----------+-------------------------------------------+-------------+
| user         | host      | password                                  | plugin      |
+--------------+-----------+-------------------------------------------+-------------+
| root         | localhost |                                           | unix_socket |
| root         | 127.0.0.1 | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| root         | ::1       | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | 127.0.0.1 | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | ::1       | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| mariaDBadmin | localhost | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| nextcloud    | 127.0.0.1 | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| nextcloud    | ::1       | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| nextcloud    | localhost | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| iiab_wp      | localhost | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
+--------------+-----------+-------------------------------------------+-------------+
10 rows in set (0.001 sec)

**Comment by jvonau** [comment]
./runrole --reinstall wordpress and ./runrole --reinstall nextcloud run to completion.

**Comment by jvonau** [comment]
I have no clue why mysql_user and mysql_db wanted login_unix_socket in the mysql role but other uses of mysql_* in other roles did not... perplexing... 

edit: the initial ansible error was complaining about missing password or ~/.my.cnf, perhaps a empty .my.cnf is all that is needed to get by that. Would require a clean machine to test that.

**Comment by tim-moody** [comment]
only risk I see is that there could be an infinite regression if root is required to create the admin user.

assume you're testing with sudo.

**Comment by jvonau** [comment]
>TASK [mysql : Start MySQL systemd service (mariadb) to permit configuration] ************************
changed: [127.0.0.1]
TASK [mysql : Install /root/.my.cnf file from template, with root password credentials] *************
changed: [127.0.0.1]
TASK [mysql : Remove the MySQL 'test' database] *****************************************************
ok: [127.0.0.1]
TASK [mysql : Delete anonymous MySQL server user for u20-srv] ***************************************
[WARNING]: Module did not set no_log for update_password
ok: [127.0.0.1]
TASK [mysql : Delete anonymous MySQL server user for localhost] *************************************
ok: [127.0.0.1]
TASK [mysql : Create MySQL root password for root accounts on (127.0.0.1, ::1)] *********************
changed: [127.0.0.1] => (item=127.0.0.1)
changed: [127.0.0.1] => (item=::1)
TASK [mysql : Set 'mysql_installed: True'] **********************************************************
ok: [127.0.0.1]

TASK [mysql : Remove the MySQL 'test' database] 
TASK [mysql : Delete anonymous MySQL server user for u20-srv]
TASK [mysql : Delete anonymous MySQL server user for localhost]
The above returned OK, these are not present and the stanzas can go away?
Do we really need to create the websockets?

**Comment by jvonau** [comment]
>MariaDB [(none)]> select user,host,password,plugin from mysql.user;
+---------------------+-----------+-------------------------------------------+-------------+
| user                | host      | password                                  | plugin      |
+---------------------+-----------+-------------------------------------------+-------------+
| root                | localhost |                                           | unix_socket |
| root                | 127.0.0.1 | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| root                | ::1       | *2052C731E27238D5CB438670528D8AFEEB3B3C42 |             |
| Admin               | 127.0.0.1 | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
| Admin               | ::1       | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
| Admin               | localhost | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
| iiab_mediawiki_user | localhost | *041FF3DE6FFF8CCE4AE95A4608CAA8FB9EB596C6 |             |
| nextcloud           | 127.0.0.1 | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| nextcloud           | ::1       | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| nextcloud           | localhost | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| iiab_wp             | localhost | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
+---------------------+-----------+-------------------------------------------+-------------+
11 rows in set (0.002 sec)

Think this looks ok.

**Comment by holta** [comment]
@jvonau is {{ mysql_root_password }} going away?

And if so (?) should we...

- Remove it from [default_vars.yml Lines 246-247](https://github.com/iiab/iiab/blob/master/vars/default_vars.yml#L246-L247) ?
- Ask @m-anish to revise Lines 54 & 65 of https://github.com/iiab/iiab/blob/master/roles/pbx/tasks/freepbx.yml#L54-L65 ?

**Comment by jvonau** [comment]
> @jvonau is {{ mysql_root_password }} going away?
> 
> And if so (?) should we...
> 
>     * Remove it from [default_vars.yml Lines 246-247](https://github.com/iiab/iiab/blob/master/vars/default_vars.yml#L246-L247) ?

done and defaults/main.yml
> 
>     * Ask @m-anish to revise Lines 54 & 65 of https://github.com/iiab/iiab/blob/master/roles/pbx/tasks/freepbx.yml#L54-L65 ?

done, and I'll add pbx into the next test install

**Comment by jvonau** [comment]
TASK [pbx : FreePBX - Install (just run once) - CAN TAKE 12 MIN OR LONGER!] *************************
changed: [127.0.0.1] => (item=./start_asterisk start)
Setting Permissions...
Setting base permissions...Done
Setting specific permissions...
    0 [>---------------------------]
In Encoding.php line 196:
                                                                         
  Array and string offset access syntax with curly braces is deprecated  
                                                                         

chown [-f|--file FILE] [-m|--module MODULE]

failed: [127.0.0.1] (item=./install -n --webroot /var/www/html/freepbx --dbuser asterisk --dbpass asterisk --dbname asterisk --cdrdbname asteriskcdrdb) => {"ansible_loop_var": "item", "changed": true, "cmd": ["./install", "-n", "--webroot", "/var/www/html/freepbx", "--dbuser", "asterisk", "--dbpass", "asterisk", "--dbname", "asterisk", "--cdrdbname", "asteriskcdrdb"], "delta": "0:00:06.534530", "end": "2020-08-24 05:11:47.592593", "item": "./install -n --webroot /var/www/html/freepbx --dbuser asterisk --dbpass asterisk --dbname asterisk --cdrdbname asteriskcdrdb", "msg": "non-zero return code", "rc": 1, "start": "2020-08-24 05:11:41.058063", "stderr": "    0/9363 [>---------------------------]   0%\n  100/9363 [>---------------------------]   1%\n  200/9363 [>---------------------------]   2%\n  300/9363 [>---------------------------]   3%\n  400/9363 [=>--------------------------]   4%\n  500/9363 [=>--------------------------]   5%\n  600/9363 [=>--------------------------]   6%\n  700/9363 [==>-------------------------]   7%\n  800/9363 [==>-------------------------]   8%\n  900/9363 [==>-------------------------]   9%\n 1000/9363 [==>-------------------------]  10%\n 1100/9363 [===>------------------------]  11%\n 1200/9363 [===>------------------------]  12%\n 1300/9363 [===>------------------------]  13%\n 1400/9363 [====>-----------------------]  14%\n 1500/9363 [====>-----------------------]  16%\n 1600/9363 [====>-----------------------]  17%\n 1700/9363 [=====>----------------------]  18%\n 1800/9363 [=====>----------------------]  19%\n 1900/9363 [=====>----------------------]  20%\n 2000/9363 [=====>----------------------]  21%\n 2100/9363 [======>---------------------]  22%\n 2200/9363 [======>---------------------]  23%\n 2300/9363 [======>---------------------]  24%\n 2400/9363 [=======>--------------------]  25%\n 2500/9363 [=======>--------------------]  26%\n 2600/9363 [=======>--------------------]  27%\n 2700/9363 [========>-------------------]  28%\n 2800/9363 [========>-------------------]  29%\n 2900/9363 [========>-------------------]  30%\n 3000/9363 [========>-------------------]  32%\n 3100/9363 [=========>------------------]  33%\n 3200/9363 [=========>------------------]  34%\n 3300/9363 [=========>------------------]  35%\n 3400/9363 [==========>-----------------]  36%\n 3500/9363 [==========>-----------------]  37%\n 3600/9363 [==========>-----------------]  38%\n 3700/9363 [===========>----------------]  39%\n 3800/9363 [===========>----------------]  40%\n 3900/9363 [===========>----------------]  41%\n 4000/9363 [===========>----------------]  42%\n 4100/9363 [============>---------------]  43%\n 4200/9363 [============>---------------]  44%\n 4300/9363 [============>---------------]  45%\n 4400/9363 [=============>--------------]  46%\n 4500/9363 [=============>--------------]  48%\n 4600/9363 [=============>--------------]  49%\n 4700/9363 [==============>-------------]  50%\n 4800/9363 [==============>-------------]  51%\n 4900/9363 [==============>-------------]  52%\n 5000/9363 [==============>-------------]  53%\n 5100/9363 [===============>------------]  54%\n 5200/9363 [===============>------------]  55%\n 5300/9363 [===============>------------]  56%\n 5400/9363 [================>-----------]  57%\n 5500/9363 [================>-----------]  58%\n 5600/9363 [================>-----------]  59%\n 5700/9363 [=================>----------]  60%\n 5800/9363 [=================>----------]  61%\n 5900/9363 [=================>----------]  63%\n 6000/9363 [=================>----------]  64%\n 6100/9363 [==================>---------]  65%\n 6200/9363 [==================>---------]  66%\n 6300/9363 [==================>---------]  67%\n 6400/9363 [===================>--------]  68%\n 6500/9363 [===================>--------]  69%\n 6600/9363 [===================>--------]  70%\n 6700/9363 [====================>-------]  71%\n 6800/9363 [====================>-------]  72%\n 6900/9363 [====================>-------]  73%\n 7000/9363 [====================>-------]  74%\n 7100/9363 [=====================>------]  75%\n 7200/9363 [=====================>------]  76%\n 7300/9363 [=====================>------]  77%\n 7400/9363 [======================>-----]  79%\n 7500/9363 [======================>-----]  80%\n 7600/9363 [======================>-----]  81%\n 7700/9363 [=======================>----]  82%\n 7800/9363 [=======================>----]  83%\n 7900/9363 [=======================>----]  84%\n 8000/9363 [=======================>----]  85%\n 8100/9363 [========================>---]  86%\n 8200/9363 [========================>---]  87%\n 8300/9363 [========================>---]  88%\n 8400/9363 [=========================>--]  89%\n 8500/9363 [=========================>--]  90%\n 8600/9363 [=========================>--]  91%\n 8700/9363 [==========================>-]  92%\n 8800/9363 [==========================>-]  93%\n 8900/9363 [==========================>-]  95%\n 9000/9363 [==========================>-]  96%\n 9100/9363 [===========================>]  97%\n 9200/9363 [===========================>]  98%\n 9300/9363 [===========================>]  99%\n 9363/9363 [============================] 100%\nIn Process.php line 239:\n                                                   \n  The command \"/usr/sbin/fwconsole chown\" failed.  \n                                                   \n  Exit Code: 255(Unknown error)                    \n                                                   \n  Working directory: /opt/iiab/freepbx             \n                                                   \n  Output:                                          \n  ================                                 \n                                                   \n                                                   \n  Error Output:                                    \n  ================                                 \n                                                   \n\ninstall [--dbengine DBENGINE] [--dbname DBNAME] [--dbhost DBHOST] [--cdrdbname CDRDBNAME] [--dbuser DBUSER] [--dbpass DBPASS] [--user USER] [--group GROUP] [--dev-links] [--skip-install] [--webroot WEBROOT] [--astetcdir ASTETCDIR] [--astmoddir ASTMODDIR] [--astvarlibdir ASTVARLIBDIR] [--astagidir ASTAGIDIR] [--astspooldir ASTSPOOLDIR] [--astrundir ASTRUNDIR] [--astlogdir ASTLOGDIR] [--ampbin AMPBIN] [--ampsbin AMPSBIN] [--ampcgibin AMPCGIBIN] [--ampplayback AMPPLAYBACK] [-r|--rootdb] [-f|--force]", "stderr_lines": ["    0/9363 [>---------------------------]   0%", "  100/9363 [>---------------------------]   1%", "  200/9363 [>---------------------------]   2%", "  300/9363 [>---------------------------]   3%", "  400/9363 [=>--------------------------]   4%", "  500/9363 [=>--------------------------]   5%", "  600/9363 [=>--------------------------]   6%", "  700/9363 [==>-------------------------]   7%", "  800/9363 [==>-------------------------]   8%", "  900/9363 [==>-------------------------]   9%", " 1000/9363 [==>-------------------------]  10%", " 1100/9363 [===>------------------------]  11%", " 1200/9363 [===>------------------------]  12%", " 1300/9363 [===>------------------------]  13%", " 1400/9363 [====>-----------------------]  14%", " 1500/9363 [====>-----------------------]  16%", " 1600/9363 [====>-----------------------]  17%", " 1700/9363 [=====>----------------------]  18%", " 1800/9363 [=====>----------------------]  19%", " 1900/9363 [=====>----------------------]  20%", " 2000/9363 [=====>----------------------]  21%", " 2100/9363 [======>---------------------]  22%", " 2200/9363 [======>---------------------]  23%", " 2300/9363 [======>---------------------]  24%", " 2400/9363 [=======>--------------------]  25%", " 2500/9363 [=======>--------------------]  26%", " 2600/9363 [=======>--------------------]  27%", " 2700/9363 [========>-------------------]  28%", " 2800/9363 [========>-------------------]  29%", " 2900/9363 [========>-------------------]  30%", " 3000/9363 [========>-------------------]  32%", " 3100/9363 [=========>------------------]  33%", " 3200/9363 [=========>------------------]  34%", " 3300/9363 [=========>------------------]  35%", " 3400/9363 [==========>-----------------]  36%", " 3500/9363 [==========>-----------------]  37%", " 3600/9363 [==========>-----------------]  38%", " 3700/9363 [===========>----------------]  39%", " 3800/9363 [===========>----------------]  40%", " 3900/9363 [===========>----------------]  41%", " 4000/9363 [===========>----------------]  42%", " 4100/9363 [============>---------------]  43%", " 4200/9363 [============>---------------]  44%", " 4300/9363 [============>---------------]  45%", " 4400/9363 [=============>--------------]  46%", " 4500/9363 [=============>--------------]  48%", " 4600/9363 [=============>--------------]  49%", " 4700/9363 [==============>-------------]  50%", " 4800/9363 [==============>-------------]  51%", " 4900/9363 [==============>-------------]  52%", " 5000/9363 [==============>-------------]  53%", " 5100/9363 [===============>------------]  54%", " 5200/9363 [===============>------------]  55%", " 5300/9363 [===============>------------]  56%", " 5400/9363 [================>-----------]  57%", " 5500/9363 [================>-----------]  58%", " 5600/9363 [================>-----------]  59%", " 5700/9363 [=================>----------]  60%", " 5800/9363 [=================>----------]  61%", " 5900/9363 [=================>----------]  63%", " 6000/9363 [=================>----------]  64%", " 6100/9363 [==================>---------]  65%", " 6200/9363 [==================>---------]  66%", " 6300/9363 [==================>---------]  67%", " 6400/9363 [===================>--------]  68%", " 6500/9363 [===================>--------]  69%", " 6600/9363 [===================>--------]  70%", " 6700/9363 [====================>-------]  71%", " 6800/9363 [====================>-------]  72%", " 6900/9363 [====================>-------]  73%", " 7000/9363 [====================>-------]  74%", " 7100/9363 [=====================>------]  75%", " 7200/9363 [=====================>------]  76%", " 7300/9363 [=====================>------]  77%", " 7400/9363 [======================>-----]  79%", " 7500/9363 [======================>-----]  80%", " 7600/9363 [======================>-----]  81%", " 7700/9363 [=======================>----]  82%", " 7800/9363 [=======================>----]  83%", " 7900/9363 [=======================>----]  84%", " 8000/9363 [=======================>----]  85%", " 8100/9363 [========================>---]  86%", " 8200/9363 [========================>---]  87%", " 8300/9363 [========================>---]  88%", " 8400/9363 [=========================>--]  89%", " 8500/9363 [=========================>--]  90%", " 8600/9363 [=========================>--]  91%", " 8700/9363 [==========================>-]  92%", " 8800/9363 [==========================>-]  93%", " 8900/9363 [==========================>-]  95%", " 9000/9363 [==========================>-]  96%", " 9100/9363 [===========================>]  97%", " 9200/9363 [===========================>]  98%", " 9300/9363 [===========================>]  99%", " 9363/9363 [============================] 100%", "In Process.php line 239:", "                                                   ", "  The command \"/usr/sbin/fwconsole chown\" failed.  ", "                                                   ", "  Exit Code: 255(Unknown error)                    ", "                                                   ", "  Working directory: /opt/iiab/freepbx             ", "                                                   ", "  Output:                                          ", "  ================                                 ", "                                                   ", "                                                   ", "  Error Output:                                    ", "  ================                                 ", "                                                   ", "", "install [--dbengine DBENGINE] [--dbname DBNAME] [--dbhost DBHOST] [--cdrdbname CDRDBNAME] [--dbuser DBUSER] [--dbpass DBPASS] [--user USER] [--group GROUP] [--dev-links] [--skip-install] [--webroot WEBROOT] [--astetcdir ASTETCDIR] [--astmoddir ASTMODDIR] [--astvarlibdir ASTVARLIBDIR] [--astagidir ASTAGIDIR] [--astspooldir ASTSPOOLDIR] [--astrundir ASTRUNDIR] [--astlogdir ASTLOGDIR] [--ampbin AMPBIN] [--ampsbin AMPSBIN] [--ampcgibin AMPCGIBIN] [--ampplayback AMPPLAYBACK] [-r|--rootdb] [-f|--force]"], "stdout": "Checking if SELinux is enabled...Its not (good)!\nReading /etc/asterisk/asterisk.conf...Done\nChecking if Asterisk is running and we can talk to it as the 'asterisk' user...Yes. Determined Asterisk version to be: 16.12.0\nChecking if NodeJS is installed and we can get a version from it...Yes. Determined NodeJS version to be: 12.18.3\nPreliminary checks done. Starting FreePBX Installation\nChecking if this is a new install...Yes (No /etc/freepbx.conf file detected)\nDatabase installation checking credentials and permissions..Connected!\nBlank /etc/odbc.ini file detected. Installing...Done\nEmpty asterisk Database going to populate it\nUpdating tables admin, ampusers, cronmanager, featurecodes, freepbx_log, freepbx_settings, globals, module_xml, modules, notifications, cron_jobs...Done\nEmpty asteriskcdrdb Database going to populate it\nInitializing FreePBX Settings\n\tChanging AMPWEBROOT [/var/www/html] to match what was given at install time: /var/www/html/freepbx\n\tChanging AMPMGRPASS [amp111] to match what was given at install time: 2503b498e870e000405d2d599755ba57\nFinished initalizing settings\nCopying files (this may take a bit)....\n\nDone\nbin is: /var/lib/asterisk/bin\nsbin is: /usr/sbin\nSymlinking /var/lib/asterisk/bin/fwconsole to /usr/sbin/fwconsole ...Done\nSymlinking /var/lib/asterisk/bin/amportal to /usr/sbin/amportal ...Done\nFinishing up directory processes...Done!\nRunning variable replacement...Done\nCreating missing #include files...Done\nSetting up Asterisk Manager Connection...Done\nRunning through upgrades...\nChecking for upgrades..\nNo further upgrades necessary\nFinished upgrades\nSetting FreePBX version to 15.0.16.60...Done\nWriting out /etc/amportal.conf...Done\nWriting out /etc/freepbx.conf...Done\nChowning directories...", "stdout_lines": ["Checking if SELinux is enabled...Its not (good)!", "Reading /etc/asterisk/asterisk.conf...Done", "Checking if Asterisk is running and we can talk to it as the 'asterisk' user...Yes. Determined Asterisk version to be: 16.12.0", "Checking if NodeJS is installed and we can get a version from it...Yes. Determined NodeJS version to be: 12.18.3", "Preliminary checks done. Starting FreePBX Installation", "Checking if this is a new install...Yes (No /etc/freepbx.conf file detected)", "Database installation checking credentials and permissions..Connected!", "Blank /etc/odbc.ini file detected. Installing...Done", "Empty asterisk Database going to populate it", "Updating tables admin, ampusers, cronmanager, featurecodes, freepbx_log, freepbx_settings, globals, module_xml, modules, notifications, cron_jobs...Done", "Empty asteriskcdrdb Database going to populate it", "Initializing FreePBX Settings", "\tChanging AMPWEBROOT [/var/www/html] to match what was given at install time: /var/www/html/freepbx", "\tChanging AMPMGRPASS [amp111] to match what was given at install time: 2503b498e870e000405d2d599755ba57", "Finished initalizing settings", "Copying files (this may take a bit)....", "", "Done", "bin is: /var/lib/asterisk/bin", "sbin is: /usr/sbin", "Symlinking /var/lib/asterisk/bin/fwconsole to /usr/sbin/fwconsole ...Done", "Symlinking /var/lib/asterisk/bin/amportal to /usr/sbin/amportal ...Done", "Finishing up directory processes...Done!", "Running variable replacement...Done", "Creating missing #include files...Done", "Setting up Asterisk Manager Connection...Done", "Running through upgrades...", "Checking for upgrades..", "No further upgrades necessary", "Finished upgrades", "Setting FreePBX version to 15.0.16.60...Done", "Writing out /etc/amportal.conf...Done", "Writing out /etc/freepbx.conf...Done", "Chowning directories..."]}

PLAY RECAP ******************************************************************************************
127.0.0.1                  : ok=79   changed=25   unreachable=0    failed=1    skipped=19   rescued=0

**Comment by jvonau** [comment]
>iiab-admin@u20-srv:/opt/iiab/iiab$ sudo mysql -u root
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 65
Server version: 10.3.22-MariaDB-1ubuntu1 Ubuntu 20.04
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
MariaDB [(none)]> select user,host,password,plugin from mysql.user;
+---------------------+-----------+-------------------------------------------+-------------+
| user                | host      | password                                  | plugin      |
+---------------------+-----------+-------------------------------------------+-------------+
| root                | localhost |                                           | unix_socket |
| Admin               | localhost | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
| iiab_mediawiki_user | localhost | *041FF3DE6FFF8CCE4AE95A4608CAA8FB9EB596C6 |             |
| nextcloud           | localhost | *A48743A94A55054A7D9C3B463815AF1BE15B4061 |             |
| iiab_wp             | localhost | *7ACE763ED393514FE0C162B93996ECD195FFC4F5 |             |
| asterisk            | localhost | *7C99FE2524A21E88E9DBE310DB2F754773BF383E |             |
+---------------------+-----------+-------------------------------------------+-------------+
6 rows in set (0.001 sec)

looks good and as a bonus pbx ran to completion

**Comment by jvonau** [comment]
https://community.freepbx.org/t/freepbx-15-0-16-44-on-aws-ubuntu-20-with-asterisk-16-failing-to-install/66985/7

**Comment by jvonau** [comment]
>iiab-admin@u20-srv:/opt/iiab/iiab$ sudo mysql -u asterisk -p
Enter password: 
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 181
Server version: 10.3.22-MariaDB-1ubuntu1 Ubuntu 20.04
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
MariaDB [(none)]> select user,host,password,plugin from mysql.user;
ERROR 1142 (42000): SELECT command denied to user 'asterisk'@'localhost' for table 'user'
MariaDB [(none)]> 

Good only 'root' can access the user table

**Comment by holta** [comment]
@jvonau has tested this on Ubuntu Server 20.04.1 VM PR2488 = 10.8.0.38

This is also now beginning to be test-installed onto RaspiOS i.e. 183-pi4-2g-lite-BIG-PR2488 = 10.8.0.22

**Comment by tim-moody** [comment]
I'm wondering why this was not sufficient:

[client]
user=root
password =
socket   = /run/mysqld/mysqld.sock

**Comment by jvonau** [comment]
> I'm wondering why this was not sufficient:
> 
> [client]
> user=root
> password =
> socket = /run/mysqld/mysqld.sock

To be consistent with the format of /etc/mysql/debian.cnf and other os supplied .cnf file present on the filesystem.

**Comment by jvonau** [comment]
> (Should this PR #2488 be merged today?)

I'd like to limit the scope to just mariadb related changes, I can rebase this PR back to 59c8ce7 and save the pbx stuff for a different PR. Sounds ok with you?

**Comment by holta** [comment]
> > (Should this PR #2488 be merged today?)
> 
> I'd like to limit the scope to just mariadb related changes, I can rebase this PR back to [59c8ce7](https://github.com/iiab/iiab/commit/59c8ce764e9d79f7f0bd5b561dd634e1abc349b8) and save the pbx stuff for a different PR. Sounds ok with you?

Testing seems largely complete as it is: can this PR perhaps be converted into 2 PR's (MariaDB + associated changes) ...and then both merged (?)

**Comment by jvonau** [comment]
I'd like @m-anish to review the pbx changes(it's not complete need to add the logic for the last patch commit to be usable), that would be the point to splitting this PR up, otherwise I see no point in creating two PRs just to have them both merged at the same time.

**Comment by holta** [comment]
Good Idea: Let me ask @m-anish to pass judgment in coming days, so the community can move forward now, in one direction or another.

**Comment by jvonau** [comment]
rolled back to 59c8ce764e9d79f7f0bd5b561dd634e1abc349b8, can merge anytime.

**Comment by holta** [comment]
This revised (slimmed down) PR #2488 is now being tested with a fresh IIAB install just begun on 192-u20desk-BIG-PR2488 = 10.8.0.54

**Comment by holta** [comment]
The above install completed.

Full login (including clicking thru each app's introductory wizard) was completed for these 3, per @jvonau's recommendation:

- http://10.8.0.54/nextcloud
- http://10.8.0.54/wordpress
- http://10.8.0.54/mediawiki

**Pull request merged**